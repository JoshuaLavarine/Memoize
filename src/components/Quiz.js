import React, { Component } from 'react';
import Score from './Score'

class Quiz extends Component {
  constructor() {
    super();
    this.state = {
      currentQuestion: 1,
      guessesCorrect: 0,
      guessValue: '',
      displayScore: false
    }
  }

  makeGuess = () => {

  }

  displayPrompt = () => {
    let selectedPioneer = this.props.pioneers.pioneersData.find(pioneer => {
      return pioneer.id == this.props.currentPioneer
    })
    let currentQuestion = selectedPioneer.multipleChoice.questions.find(question => {
      return question.id == this.state.currentQuestion
    })





    
    return (
      <section>
        <p>Question: {currentQuestion.prompt}</p>
        <p>Possible Answers:</p>
      </section>
    )
  }

  displayPossibleAnswers = () => {
    let selectedPioneer = this.props.pioneers.pioneersData.find(pioneer => {
      return pioneer.id == this.props.currentPioneer
    })
    let currentQuestion = selectedPioneer.multipleChoice.questions.find(question => {
      return question.id == this.state.currentQuestion
    })
    let incorrectChoices = currentQuestion.incorrectAnswers.map(question => {
      return (
        <section>
          <input type="radio" id={Date.now()} value={question}></input>
          <label for={Date.now()}>{question}</label>
        </section>
      )
    })
    return incorrectChoices
  }

  displayCorrectAnswer = () => {
    let selectedPioneer = this.props.pioneers.pioneersData.find(pioneer => {
      return pioneer.id == this.props.currentPioneer
    })
    let currentQuestion = selectedPioneer.multipleChoice.questions.find(question => {
      return question.id == this.state.currentQuestion
    })
    let correctAnswer = currentQuestion.correctAnswer
    return (
      <section>
      <input type="radio" id={Date.now()} value={correctAnswer}></input>
      <label for={Date.now()}>{correctAnswer}</label>
      </section>
    )
  }

  getClickedValue = (event) => {
    this.setState({
      guessValue: event.target.value
    })
  }

  checkCorrectAnswer = (e) => {
    e.preventDefault();
    let selectedPioneer = this.props.pioneers.pioneersData.find(pioneer => {
      return pioneer.id == this.props.currentPioneer
    })
    let currentQuestion = selectedPioneer.multipleChoice.questions.find(question => {
      return question.id == this.state.currentQuestion
    })
    let correctAnswer = currentQuestion.correctAnswer
    if (this.state.guessValue === correctAnswer) {
      this.setState({
        guessesCorrect: this.state.guessesCorrect + 1,
        currentQuestion: this.state.currentQuestion + 1,
        guessValue: '',
      })
      this.displayScore()
    } else {
      this.setState({
        currentQuestion: this.state.currentQuestion + 1,
        guessValue: '',
      })
      this.displayScore()
    }
  }

  displayScore = () => {
    this.setState({
      displayScore: !this.state.displayScore
    })

  }

  render() {
    switch(this.state.displayScore) {
      case(true):
      return (
          <Score
          currentQuestion = {this.state.currentQuestion}
          guessesCorrect = {this.state.guessesCorrect}
          displayScore = {this.displayScore}
          />
      )
      default:
        return (
          <form>
            <section onClick={this.getClickedValue}>{this.displayPrompt()}</section>
            <section onClick={this.getClickedValue}>{this.displayCorrectAnswer()}</section>
            <section onClick={this.getClickedValue}>{this.displayPossibleAnswers()}</section>
            <button type="reset" onClick={this.checkCorrectAnswer}>Submit Answer</button>
          </form>
        )
    }
  }
}

export default Quiz;