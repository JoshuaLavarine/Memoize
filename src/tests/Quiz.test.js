import React from "react";
import Quiz from "../Components/Quiz";
import { shallow } from "enzyme";
import { pioneers } from "./mockData";

describe("Quiz", () => {
  const mockSelectPioneer = jest.fn();
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Quiz 
      currentPioneer={1}
      pioneers={pioneers}
      selectPioneer = {mockSelectPioneer}
      incorrectAnswers = {{
        1: [1, 2, 3, 4, 5],
        2: [1, 2, 3, 4, 5],
        3: [1, 2, 3, 4, 5],
        4: [1, 2, 3, 4, 5],
        5: [1, 2, 3, 4, 5]
      }}
      />   
    );
  });
	
  it("should match the snapshot with all data passed in", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have a proper default state", () => {
    expect(wrapper.state()).toEqual({ 
      currentUserGuess: 0,
      currentQuestionIndex: 0,
      guessesCorrect: 0,
      guessValue: '',
      displayScore: false,
      enableButton: true
		});
	});
});