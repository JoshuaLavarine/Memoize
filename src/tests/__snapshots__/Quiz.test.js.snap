// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz should match the snapshot with all data passed in 1`] = `
<form>
  <section
    onClick={[Function]}
  >
    <section>
      <p>
        Question: 
        At which university did Grace Hopper complete her Ph.D?
      </p>
      <p>
        Possible Answers:
      </p>
    </section>
  </section>
  <section
    onClick={[Function]}
  >
    <section>
      <label
        className="radio-buttons"
        onClick={[Function]}
      >
        <input
          id={1551140717394}
          name="radioBtns"
          onClick={[Function]}
          type="radio"
          value="Yale"
        />
        Yale
      </label>
    </section>
  </section>
  <section
    onClick={[Function]}
  >
    <section>
      <label
        className="radio-buttons"
        onClick={[Function]}
      >
        <input
          id={1551140717394}
          name="radioBtns"
          onClick={[Function]}
          type="radio"
          value="Harvard"
        />
        Harvard
      </label>
    </section>
    <section>
      <label
        className="radio-buttons"
        onClick={[Function]}
      >
        <input
          id={1551140717394}
          name="radioBtns"
          onClick={[Function]}
          type="radio"
          value="Princeton"
        />
        Princeton
      </label>
    </section>
    <section>
      <label
        className="radio-buttons"
        onClick={[Function]}
      >
        <input
          id={1551140717394}
          name="radioBtns"
          onClick={[Function]}
          type="radio"
          value="Louisiana State University"
        />
        Louisiana State University
      </label>
    </section>
  </section>
  <section
    className="button-container"
  >
    <button
      className="button-back"
      onClick={[MockFunction]}
      value={0}
    >
      Return to Home Screen
    </button>
    <button
      className="button-quiz"
      disabled={true}
      onClick={[Function]}
    >
      Submit Answer
    </button>
  </section>
</form>
`;
